### PR TITLE
CMake: replaced obsolete use of add_compiler_export_flags() function (#72)

### PR DESCRIPTION
**Pull request by agarny**
Addresses issue #72.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -65,7 +65,8 @@ set(HEADER_FILES
   ${LIBCELLML_CONFIG_H}
 )
 
-add_compiler_export_flags()
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
```
Comment (hsorby) [inline src/CMakeLists.txt]: These values should be set as properties on the target not through manipulating the default CMake variables.

Comment (agarny) [inline src/CMakeLists.txt]: Good point. I believe this should now be the case with https://github.com/cellml/libcellml/pull/73/commits/d7aaf9346f7952691709be0aa8571637314f0e59.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -65,16 +65,17 @@ set(HEADER_FILES
...
   ${SOURCE_FILES}
   ${HEADER_FILES}
   ${MODULE_HEADER_FILES}
   ${API_HEADER_FILES}
 )
 
+set_target_properties(cellml PROPERTIES
+  CXX_VISIBILITY_PRESET hidden
+  VISIBILITY_INLINES_HIDDEN 1)
+
```
Comment (hsorby) [inline src/CMakeLists.txt]: Last thing and then I'm done, just a little further down in this file there is a set_target_properties command these two occurences should just be one.

Comment (agarny) [inline src/CMakeLists.txt]: Fair point. Done.

**Pull request merged**